src/wolfio.c

```diff
@@ -2895,7 +2903,11 @@ int GNRC_GenerateCookie(WOLFSSL* ssl, byte *buf, int sz, void *_ctx)
         token_size = GNRC_MAX_TOKEN_SIZE;
     XMEMSET(token, 0, GNRC_MAX_TOKEN_SIZE);
     XMEMCPY(token, &ctx->peer_addr, token_size);
+#if defined(NO_SHA) && !defined(NO_SHA256)
+    ret = wc_Sha256Hash(token, token_size, digest);
```

Comment:
Since digest is WC_SHA_DIGEST_SIZE would this right over the end of the buffer? Writing out WC_SHA256_DIGEST_SIZE.

Reply:
This would be applicable for the other spots too where SHA256 is being used in this pull request.

Reply:
really good eye -- thanks for catching that.  the first instance I fixed had
```
    byte dgt[WC_MAX_DIGEST_SIZE];
```
but the others indeed use `WC_SHA_DIGEST_SIZE`.

I'm adding sanitizer coverage of `--disable-sha` -- I made a few tweaks in `configure.ac` to get `--enable-all-crypto --disable-sha` to build and `make check` cleanly, and in the multi-test scenario I had to add `--disable-ocspstapling` because of hardcoded SHA1 in the OCSP testing, similar to the gating I added yesterday around the OCSP tests in api.c with baked-in SHA1 responses.

```
[all-crypto-intelasm-no-sha-sanitizer] [1 of 1] [eaa66dc117-dirty]
    autogen.sh eaa66dc117-dirty...   real 0m16.434s  user 0m14.496s  sys 0m1.075s
    configure...   real 0m10.831s  user 0m5.509s  sys 0m6.239s
    build...   real 1m16.426s  user 9m52.645s  sys 0m9.770s
    check...   real 0m15.955s  user 0m20.930s  sys 0m10.125s
    all-crypto-intelasm-no-sha-sanitizer OK
```

Note that it doesn't detect the stack underallocations you identified, because they have hardware-specific gates that aren't tested here.